src/types/register.ts

```diff
@@ -0,0 +1,12 @@
+import { Profile } from '@/types/profile'
+
+export interface Register {
+  // profiles: Profile[]
+}
+
+export type RegisteredProfiles = Register extends { profiles: infer TProfiles extends Readonly<Profile[]> }
+  ? TProfiles
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  : any
```

Comment:
Yes because `any` means that if you do not override Register, you basically just get `map(string, unknown, string)` so you're free to try mapping whatever you want. Empty array would not let you map anything without ts error.

Reply:
hmm right, makes sense 👍